app/src/main/java/nl/tudelft/pixelperfect/pixelperfect/mini_game/AsteroidImpactActivity.java

```diff
@@ -10,10 +22,87 @@
  * @author Floris Doolaard
  */
 public class AsteroidImpactActivity extends AppCompatActivity  {
+    private Spaceship ship = Spaceship.getInstance();
+    private GameClient game = GameClient.getInstance();
+    private ProgressBar progressBarEnergyShield;
+    private ProgressBar progressBarHyperdrive;
+    private Boolean repairingEnergyShield;
 
+    /**
+     * On creation of the Activity the the content views will be initialized.
+     *
+     * @param savedInstanceState the instance to create.
+     */
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_mini_game_asteroid_impact);
+
+        progressBarEnergyShield = (ProgressBar) findViewById(R.id.mini_game_asteroid_impact_progressBar1);
+        progressBarHyperdrive = (ProgressBar) findViewById(R.id.mini_game_asteroid_impact_progressBar2);
+
+        repairingEnergyShield = true;
+    }
+
+    /**
+     * When the user presses the button, progress should be made.
+     *
+     * @param view the View of the button.
+     */
+    public void createProgress(View view) {
+        if(repairingEnergyShield){
+            increaseProgress(progressBarEnergyShield);
+        } else {
+            increaseProgress(progressBarHyperdrive);
+        }
+    }
+
+    /**
+     * Increases progress of a progressBar and sends a message whenever it is done.
+     *
+     * @param progressBar the progressBar on which progress must increase.
+     */
+    public void increaseProgress(ProgressBar progressBar){
+        if(progressBar.getProgress() == 95) {
+            if(ship.getEventLog().contains(Events.ASTEROID)){
+                EventCompletedMessage message = new EventCompletedMessage("Asteroid Impact Event", ship.getEventLog().pop(Events.ASTEROID).getId());
+                Map<String, Integer> parameters = new HashMap<String, Integer>();
+                if(repairingEnergyShield){
+                    parameters.put("Asteroid Impact", 0);
+                } else {
+                    parameters.put("Asteroid Impact", 1);
+                }
+                message.setParameters(parameters);
+                game.sendMessage(message);
+            } else {
+                game.sendMessage(new EventCompletedMessage("WRONG ANSWER", -1));
+            }
+        }
+
+        if(progressBar.getProgress() != 100) {
+            progressBar.incrementProgressBy(5);
+        }
+    }
+
+    /**
+     * Whenever a radio button from the RadioGroup gets clicked it must select the accompanied
+     * progress bar.
+     *
+     * @param view the view of the RadioGroup.
+     */
+    public void onRadioButtonClicked(View view) {
```

Comment:
Nice, but instead of updating the state of the Activity class for each interaction with the form, you could make a single method that uses `R` to grab all the values entered in the different input fields (just like submitting an HTML form). Have you considered this?